shared/utils/graphql.ts

```diff
@@ -0,0 +1,273 @@
+import type {
+  AssertionGraphEntity,
+  ChainId,
+  ChainName,
+  OOV2GraphEntity,
+  OracleType,
+  ParsedOOV1GraphEntity,
+  ParsedOOV2GraphEntity,
+  PriceRequestGraphEntity,
+  RequestState,
+} from "@shared/types";
+import { BigNumber } from "ethers";
+
+export function handleGraphqlNullableStringOrBytes<
+  EntityKey extends { [key: string]: string | null }
+>(entityKey: EntityKey | null) {
+  if (entityKey === null) return null;
+
+  const [property, stringOrBytes] = Object.entries(entityKey)[0];
+
+  if (!(typeof stringOrBytes === "string" || stringOrBytes === null)) {
+    console.error(
+      `Wrong type in \`handleGraphqlNullableStringOrBytes\`: Expected string or null for ${property}, got ${typeof stringOrBytes} instead`
+    );
+  }
+  if (stringOrBytes === null) return null;
+
+  return stringOrBytes;
+}
+
+export function handleGraphqlNullableBigInt<
+  EntityKey extends { [key: string]: string | null }
+>(entityKey: EntityKey | null) {
+  if (entityKey === null) return null;
+
+  const [property, bigInt] = Object.entries(entityKey)[0];
+
+  if (!(typeof bigInt === "string" || bigInt === null)) {
+    console.error(
+      `Wrong type in \`handleGraphqlNullableBigInt\`: Expected string or null for ${property}, got ${typeof bigInt} instead`
+    );
+  }
+  if (bigInt === null) return null;
+
+  return BigNumber.from(bigInt);
+}
+
+export function handleGraphqlNullableBoolean<
+  EntityKey extends { [key: string]: boolean | null }
+>(entityKey: EntityKey | null) {
+  if (entityKey === null) return null;
+
+  const [property, boolean] = Object.entries(entityKey)[0];
+
+  if (!(typeof boolean === "boolean" || boolean === null)) {
+    console.error(
+      `Wrong type in \`handleGraphqlNullableBoolean\`: Expected boolean or null for ${property}, got ${typeof boolean} instead`
+    );
+  }
+
+  if (boolean === null) return null;
+
+  return boolean;
+}
+
+export function makeQueryName(oracleType: OracleType, chainName: ChainName) {
```

Comment:
by using this util function, we can also generate query names in the mocks and have them match exactly